### PR TITLE
add getOwnPropertyDescription Entry

### LINKED ISSUE #4060: [Term Entry] JavaScript Objects .getOwnPropertyDescriptors()
### Reason/inspiration (optional)

We would like a new term entry on the `Object.getOwnPropertyDescriptors()` method for [Objects](https://www.codecademy.com/resources/docs/javascript/objects): getOwnPropertyDescriptors. The entry should go in a new file under `docs/content/javascript/concepts/objects/terms/getOwnPropertyDescriptors/getOwnPropertyDescriptors.md`.

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo` code snippet.
- An "Example" section with a ` ```js `code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Brag2gr8**
### Description

Added getOwnPropertyDescription for Javascript Object terms. closes #4060 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/javascript/concepts/objects/terms/getOwnPropertyDescriptors/getOwnPropertyDescriptors.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
+Title: '.getOwnPropertyDescriptors'
```
Comment (avdhoottt) [inline content/javascript/concepts/objects/terms/getOwnPropertyDescriptors/getOwnPropertyDescriptors.md:2]:
```suggestion
Title: 'Object.getOwnPropertyDescriptors()'
```
Use this format whenever you write the name of the method.

File: content/javascript/concepts/objects/terms/getOwnPropertyDescriptors/getOwnPropertyDescriptors.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+Description: 'returns all own property descriptors of a given object'
```
Comment (avdhoottt) [inline content/javascript/concepts/objects/terms/getOwnPropertyDescriptors/getOwnPropertyDescriptors.md:3]:
```suggestion
Description: 'Returns all own property descriptors of a given object.'
```

File: content/javascript/concepts/objects/terms/getOwnPropertyDescriptors/getOwnPropertyDescriptors.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+Tags:
+  - 'Properties'
+  - 'Objects'
+  - 'Functions'
+CatalogContent:
+  - 'introduction-to-javascript'
+  - 'paths/create-a-back-end-app-with-javascript'
+---
+
+The **`Object.getOwnPropertyDescriptors()`** method returns all own property descriptors of a given object. It is useful for extracting property descriptors, including configurable, enumerable, value, and writable properties, from an object.manipulation.
```
Comment (avdhoottt) [inline content/javascript/concepts/objects/terms/getOwnPropertyDescriptors/getOwnPropertyDescriptors.md:17]:
```suggestion
The **`Object.getOwnPropertyDescriptors()`** method returns all own property descriptors of a given object. It is useful for extracting property descriptors, including configurable, enumerable, value, and writable properties, allowing manipulation of these property attributes from an object.
```

File: content/javascript/concepts/objects/terms/getOwnPropertyDescriptors/getOwnPropertyDescriptors.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+## Syntax
+
+```pseudo
+Object.getOwnPropertyDescriptors(obj)
+```
+
+`obj` is the given obeject.
```
Comment (avdhoottt) [inline content/javascript/concepts/objects/terms/getOwnPropertyDescriptors/getOwnPropertyDescriptors.md:25]:
```suggestion
- `obj`: It is the given object.
```

File: content/javascript/concepts/objects/terms/getOwnPropertyDescriptors/getOwnPropertyDescriptors.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+## Examples
```
Comment (avdhoottt) [inline content/javascript/concepts/objects/terms/getOwnPropertyDescriptors/getOwnPropertyDescriptors.md:27]: Remove this heading and add the rest of the two (`Example 1` & `Example 2`) headings as h2.

File: content/javascript/concepts/objects/terms/getOwnPropertyDescriptors/getOwnPropertyDescriptors.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+The following codes demonstrate a basic implementation of the `Object.getOwnPropertyDescriptors()` method:
+
+### Example 1
+
+In this example, Object.getOwnPropertyDescriptors() returns an object where each key is a property name, and the corresponding value is the property descriptor for that property.
```
Comment (avdhoottt) [inline content/javascript/concepts/objects/terms/getOwnPropertyDescriptors/getOwnPropertyDescriptors.md:33]:
```suggestion
In this example, `Object.getOwnPropertyDescriptors()` returns an object where each key is a property name, and the corresponding value is the property descriptor for that property.
```

File: content/javascript/concepts/objects/terms/getOwnPropertyDescriptors/getOwnPropertyDescriptors.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+```js
+const obj = {
+  name: 'John',
+  age: 25,
+};
+
+const propertyDescriptors = Object.getOwnPropertyDescriptors(obj);
+
```
Comment (avdhoottt) [inline content/javascript/concepts/objects/terms/getOwnPropertyDescriptors/getOwnPropertyDescriptors.md:42]:
```suggestion
```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@Brag2gr8 Nice Work! Here are some changes from my side, please make them. I might have some more changes in the future.

**Comment by Brag2gr8** [comment]
@avdhoottt I'm done implementing the proposed changes

**Review by avdhoottt — APPROVED** [review]
Good to go for the second review.

**Review by dakshdeepHERE — APPROVED** [review]
This looks good to merge 🚀

**Pull request merged**